csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -192,6 +226,31 @@ public static string ToDiagnosticString(IMessage message)
             return diagnosticFormatter.Format(message);
         }
 
+        /// <summary>
+        /// Converts the field of a message to JSON for diagnostic purposes with no extra context.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This differs from calling <see cref="Format(IMessage, FieldDescriptor)"/> on the default JSON
+        /// formatter in its handling of <see cref="Any"/>. As no type registry is available
+        /// in <see cref="object.ToString"/> calls, the normal way of resolving the type of
+        /// an <c>Any</c> message cannot be applied. Instead, a JSON property named <c>@value</c>
```

Comment:
Presumably the results can differ for anything implementing ICustomDiagsnoticMessage too?
If this was documentation copied from another method, we should probably fix both. (I can do that later though.)